### ISSUE TITLE
Ping-pong between C400 and C416

### ISSUE THREAD

**Issue by jenstroeger** [adamchainz/flake8-comprehensions#643]
### Python Version

3.13.12

### flake8 Version

7.3.0

### Package Version

3.17.0

### Description

Hi,  with this line of code (slightly modified here)
```python
node["children"] = [n for n in f(node["children"])]
```
where `f()` yields the elements of a new list, I get 
```
path/to/file.py:1853:48: C416 Unnecessary list comprehension - rewrite using list().
                        node["children"] = [n for n in f(node["children"])]
```
and when I rewrite the code accordingly I get
```
path/to/file.py:1853:48: C400 Unnecessary generator - rewrite as a list comprehension.
                        node["children"] = list(n for n in f(node["children"]))
```
I’m going to stick with the comprehension and silence this warning, but this should probably be fixed?
